lib/pos_block_producer.go

```diff
@@ -111,10 +145,77 @@ func (pbp *PosBlockProducer) createBlockTemplate(latestBlockView *UtxoView, newB
 	return block, nil
 }
 
+func (pbp *PosBlockProducer) getMockBlockSignature() (*bls.Signature, error) {
+	if pbp.mockBlockSignature != nil {
+		return pbp.mockBlockSignature, nil
+	}
+	mockBLSPrivateKey, err := bls.NewPrivateKey()
+	if err != nil {
+		return nil, errors.Wrap(err, "Error creating mock BLS private key")
+	}
+	mockBLSSigner, err := NewBLSSigner(mockBLSPrivateKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error creating mock BLSSigner")
+	}
+	mockBlockSignature, err := mockBLSSigner.SignBlockProposal(math.MaxUint64, NewBlockHash(RandomBytes(32)))
+	if err != nil {
+		return nil, errors.Wrap(err, "Error creating mock block signature")
+	}
+	pbp.mockBlockSignature = mockBlockSignature
+	return mockBlockSignature, nil
+}
+
+func (pbp *PosBlockProducer) mockHeader(
+	latestBlockView *UtxoView,
+	newBlockHeight uint64,
+	view uint64,
+	proposerRandomSeedSignature *bls.Signature,
+	validatorsVoteQC *QuorumCertificate,
+	validatorTimeoutAggregateQC *TimeoutAggregateQuorumCertificate,
+) (*MsgDeSoHeader, error) {
```

Comment:
This is great. Only suggestion is to encapsulate all of the mock header size computation within this function. One risk with exposing the mocked header outside of this function is that you then have a mocked MsgDeSoHeader instance laying around that could be mistaken for the true header. So you may as well encapsulate all of its usage here and expose the mocked header size instead
```suggestion
// estimateHeaderSize computes the MsgDeSoHeader message size resulting from the given params. It's 
// used to estimate the header size for block size computations when the true block header has not been
// constructed yet
func (pbp *PosBlockProducer) estimateHeaderSize(
	latestBlockView *UtxoView,
	newBlockHeight uint64,
	view uint64,
	proposerRandomSeedSignature *bls.Signature,
	validatorsVoteQC *QuorumCertificate,
	validatorTimeoutAggregateQC *TimeoutAggregateQuorumCertificate,
) (uint64, error) {
```